torchtune/_cli/download.py

```diff
@@ -159,3 +202,82 @@ def _download_cmd(self, args: argparse.Namespace) -> None:
             *list(Path(true_output_dir).iterdir()),
             sep="\n",
         )
+
+    def _download_from_kaggle(self, args: argparse.Namespace) -> None:
+        """Downloads a model from the Kaggle Model Hub."""
+
+        # Note: Kaggle doesn't actually use the "repository" terminology, but we still reuse args.repo_id here for simplicity
+        model_handle = args.repo_id
+        self._validate_kaggle_model_handle(model_handle)
+
+        # kagglehub doesn't currently support `local_dir` and `ignore_patterns` like huggingface_hub
+        if args.output_dir is not None:
+            warn(
+                "--output-dir flag is not supported for Kaggle model downloads. "
+                "This argument will be ignored."
+            )
+
+        if args.kaggle_username is not None and args.kaggle_api_key is not None:
+
+            set_kaggle_credentials(args.kaggle_username, args.kaggle_api_key)
+        elif args.kaggle_username is not None and args.kaggle_api_key is None:
+            self._parser.error(
```

Comment:
Could we check to see if the API key is defined in an environment variable here and allow it to pass through if so?

Reply:
Can do! 

passing, non-blocking comment: I initially opted not to do something like this since the correct env var to use is a bit of a hidden global implementation detail in `kagglehub`. Wasn't sure if it would feel a little mysterious just reading it from `torchtune`'s context.

Reply:
Resolved here: c2dee3a6911a48fa718fb40face90aa190cabb9b